Many new features

mavezeau opened the pull request:
Introduce the getFields to specified only the needed field and/or received more field that the standard query 

Example:  Extract Properties inside a list<Build>

Many fixes and code cleanups

mavezeau committed:
Added support for ParentProject and Projects inside a Project

Added support for ParentProject and Projects inside a Project

mavezeau committed:
Added support to Create, move, copy a project, Support to get the artifacts filtered and detection of many new object

* Change each call of TeamCityCaller to use the interface
ITeamCityCaller
* Added new method to download artifact filtered in BuildArtifacts
(DownloadFiltered)
* Added new method to modify a build trigger in
BuildConfigs(ModifTrigger)
* Added new method to modify a build snapshot dependencies in
BuildConfigs(ModifSnapshotDependencies)
* Added new method to modify a build artifact dependencies in
BuildConfigs(ModifArtifactDependencies)
* Added new method to create a project to inherit a project (Project)
* Added new method to move a project to inherit a project (Project)
* Added new method to modify a parameters inside a project (Project)
* Added new method to  modify a setting inside a project (Project)
* Added new method to detect if the current user is Administrator (User)
* Added new method to generate a ProjectId (Since 8.0) valid for
teamcity (Project)
* Added new type of Content GetBoolean (TeamCityCaller)
* New Object detected in ArtifactDependency - SourceBuildType
* New objects detected in Build - BuildType - Properties - Running_info
* New Objects detected in BuildConfig - Number - Personnel - History -
Pinned -Running - Template
* New object detected in Project - Template
* New objects detected in SnapshotDependency - Type - SourceBuildType
* New class object Running_info - SourceBuildType - Template - Templates
* Fix some issues

nheilbuth committed:
Changed Json encoder to use lowercase

nheilbuth committed:
Changed Post<T> method to accept object as data, to utilize jsonEncoder

nheilbuth committed:
Changed Projects.Create method to use Json and NewProjectDescription

nheilbuth committed:
Added NewProjectDescription and added Project.Create with parent project

nheilbuth committed:
Changed ParentProjectWrapper to produce correct Json data

nheilbuth committed:
Changed lowercase serialization to camelcase

nheilbuth committed:
Added GenerateId from mavezeau's pull request #60

nheilbuth committed:
Added options to include/not include disconnected and unauthorized agents

nheilbuth committed:
Added Build.ById method

mavezeau committed:
Sync nheilbuth

mavezeau committed:
Added missing update

mavezeau committed:
Added support for BuildArtifactDependencies and BuildArtifactDependencies.cs into the Build object

nheilbuth committed:
Changed post and put to use data object instead of data.ToString

mavezeau committed:
TITLE:  Modify the method DownloadFiltered to use the filename and not only the extension filteredExtensionFiles => filteredFiles

DESCRIPTION:
- Modify the method DownloadFiltered to use the filename and not only the extension filteredExtensionFiles => filteredFiles
  - This method use now the wilcard to find the file (*.dll to choose all dlls)
- added new support to disable the throw exception when the Authenticate didn't work (throwExceptionOnHttpError = true (default value))

nheilbuth committed:
Added direct support for setting artifact dependencies, snapshot dependencies and build triggers

nheilbuth committed:
Adjustments to json output of artifactdependencies, snapshotdependencies and buildtriggers

ewilde committed:
Implement statistics for builds

Allows retrieval of properties such as no. of passed tests in a given
build. From
http://confluence.jetbrains.com/display/TCD8/REST+API#RESTAPI-Statistics

mavezeau committed:
Adde method inside BuildConfigs

mavezeau committed:
Merge remote-tracking branch 'remotes/upstream/pr/82'

Conflicts:
	src/TeamCitySharp/ActionTypes/IBuildConfigs.cs
	src/TeamCitySharp/ActionTypes/Projects.cs
	src/TeamCitySharp/TeamCitySharp.csproj

mavezeau committed:
Keep backwork compatibility fonction

nheilbuth committed:
Fixed snapshot dependency property types

mavezeau committed:
Commit new stock

mavezeau committed:
Added new support of investigation and test

DESCRIPTION:
Added new support of investigation, template and test

Fix for TeamCity 9 .tcbuildtag to .tcbuildid

BuildConfig:
Added EscapeDataString on the ByProjectIdAndConfigurationName method
Added new method Copy to copy a build config
Added new method CopyTemplate to copy a template only.
Added new method AttachTemplate / DetachTemplate to attach/ detach a config to template

Build:
Added new method RunningByBuildConfigId to extract the running build by buildConfig
Added new method AllRunningBuild to extract all running build

Projects:
Added new method DeletedById to delete a project by Id
Fix method copy and added support to specify a root parent

ServerInformation:
Fix getBackupStatus method

SnapshotDependency:
Fix to string return

BuildInvestigations:
Added new class to extract build investigation on build failure

TEST:
Manual test

mavezeau committed:
Introduce new GetFields method for choose needed fields  on a query.

Use fields specializations: Extract complex objects for specified Fields

Sample:
    // For each builds get only the Id, Number, Status and StartDate
    var buildField = BuildField.WithFields(id: true,number:true, status: true, startDate: true);
    var buildsFields = BuildsField.WithFields( buildField: buildField);
    var currentListBuild = client.Builds.GetFields(buildsFields.ToString()).ByBuildConfigId(currentProjectId)

Update nuget package

and changing tab spacing to 2 spaces.

Changing the assembly to 1.0.0

mavezeau committed:
Merge

mavezeau committed:
Bad merge ...

mavezeau committed:
Fix

mavezeau committed:
Try to fix merge...

mavezeau committed:
Merge branch 'master' of https://github.com/mavezeau/TeamCitySharp

mavezeau committed:
Merge branch 'master' of https://github.com/nheilbuth/TeamCitySharp

# Conflicts:
#	src/TeamCitySharp/ActionTypes/Agents.cs
#	src/TeamCitySharp/ActionTypes/BuildConfigs.cs
#	src/TeamCitySharp/ActionTypes/Builds.cs
#	src/TeamCitySharp/ActionTypes/IAgents.cs
#	src/TeamCitySharp/ActionTypes/IBuildConfigs.cs
#	src/TeamCitySharp/ActionTypes/IBuilds.cs
#	src/TeamCitySharp/ActionTypes/Projects.cs
#	src/TeamCitySharp/Connection/ITeamCityCaller.cs
#	src/TeamCitySharp/Connection/TeamCityCaller.cs
#	src/TeamCitySharp/Connection/TeamcityJsonEncoderDecoderConfiguration.cs
#	src/TeamCitySharp/DomainEntities/ArtifactDependency.cs
#	src/TeamCitySharp/DomainEntities/BuildTrigger.cs
#	src/TeamCitySharp/DomainEntities/NewProjectDescription.cs
#	src/TeamCitySharp/DomainEntities/ParentProjectWrapper.cs
#	src/TeamCitySharp/DomainEntities/Properties.cs
#	src/TeamCitySharp/DomainEntities/Property.cs
#	src/TeamCitySharp/DomainEntities/SnapshotDependency.cs
#	src/TeamCitySharp/DomainEntities/SourceBuildType.cs
#	src/TeamCitySharp/Locators/ProjectLocator.cs
#	src/TeamCitySharp/TeamCitySharp.csproj

mavezeau committed:
Cleanup code ...

mavezeau commented:
Tested with teamcity 9.1.4

mavezeau committed:
Fix unit test for nunit 3.0.1

mavezeau committed:
Added new method NextBuilds

/// <summary>
/// Retrieves the list of build after a build id.
///
/// IMPORTANT NOTE: The list starts from the latest build to oldest
(Descending)
/// </summary>
/// <param name="buildid"></param>
/// <param name="count"></param>
/// <returns></returns>
public List<Build> NextBuilds(string buildid, int count = 100)

gerads commented:
Any chance anyone is taking a look at this?

borismod commented:
I'm surprised that somebody submits a PR to this repository. I haven't seen any activity here recently, There are a few other active forks and also [FluentTc](https://github.com/QualiSystems/FluentTc) project, which I suggest you to take a look at. It supports many cool features such as custom retrieve of additional  fields, paging and has fluent API.

mavezeau committed:
Added many fields and object on the DomainEntities

Delete Package nuget inside solution

Build:
 -Added to fetch  AffectedProject

New DomainEntities:
 - Revision
 - Revisions
 - Tag
 - Tags
 - Triggered

New Fields:
 - ChangeField
 - ChangesField
 - FileField
 - FilesField
 - LastChangesField
 - RevisionField
 - RevisionsField
 - TagField
 - TagsField
 - TriggeredField
 - UserField
 - VcsRootInstanceField

And many fix...

mavezeau committed:
[TeamCitySharp] - Import internal stuff to Github and starting to publish nuget package on nuget.org

DESCRIPTION:
Import internal stuff to Github and starting to publish nuget package on nuget.org
Add missing fields
Replace SourceBuildType by BuildConfig
Use latest version of EasyHttp 1.7.0
TEST:
manual tests

REVIEWER:

REFERENCES:

mavezeau committed:
[TeamCitySharp] - Fix readme only

DESCRIPTION:

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamcitySharp] - Upgrade version 1.0.54
Code cleanup and add support to  fetch ArtifactDenpdencies and SnapshotDependencies from buildType
Use new feature in VS2015 (String interpolation)

DESCRIPTION:
Now it's possible to use the official package TeamCitySharp-forked-mavezeau on nuget.org
https://www.nuget.org/packages/TeamCitySharp-forked-mavezeau/

TEST:
Fix broken tests and add new app.config for the test to define server and username we need to use to run integration  tests

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamcitySharp] - Fix issue found by satyaprakashv

DESCRIPTION:
Bad return type after to insert a build step
Added missing fields Inherited for build Trigger and property

TEST:
Manual test

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamCitySharp] - Fix issue with Vcs creation
DESCRIPTION:
replace new ConventionResolverStrategy(ConventionResolverStrategy.WordCasing.Lowercase, "-") by new JsonResolverStrategy() to create custom json  (required to  modify all DomainEntities)
Add missing field for Vcs

Add new method:
VcsRoot:
 - VcsRoot CreateVcsRoot(VcsRoot configurationName, string projectId);
 - void SetConfigurationProperties(VcsRoot vcsRootId, string key, string value);
 - void DeleteProperties(VcsRoot vcsRootId, string parameterName);
 - void DeleteVcsRoot(VcsRoot vcsRoot);

Modified  method:
BuildConfig:
  - (JsonResolverStrategy modification) bool ModifTrigger(string buildTypeId, string triggerId, string newBt)
  - (JsonResolverStrategy modification) bool ModifSnapshotDependencies(string buildTypeId, string dependencyId, string newBt)
  - (JsonResolverStrategy modification) bool ModifArtifactDependencies(string buildTypeId, string dependencyId, string newBt)

VcsRoot:
  - (Fix) void SetVcsRootValue(VcsRoot vcsRoot, VcsRootValue field, object value)

TEST:
Manual test

REVIEWER:

Commit:
Added support for getting build queue

Commit:
Attempt to align with coding style

Commit:
Another attempt to a align with coding style

Commit:
Aligned with coding style

mavezeau committed:
Merge pull request #2 from hugener/master

Build Queue Support

mavezeau committed:
[TeamcitySharp] -  Added missing documentation after previous merge

DESCRIPTION:
thanks for your contribution Hugener

TEST:
NA

REVIEWER:

REFERENCES:

mavezeau committed:
[TeamcitySharp] - Added support of ProjectFeature

DESCRIPTION:
Added support of ProjectFeature
Added field for ProjectFeature

TEST:
Create and run test

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

vmati committed:
Update README.md

mavezeau committed:
Merge pull request #4 from vmati/patch-1

Update README.md

mavezeau committed:
[TeamCitySharp] - Add missing Inherited flag for ArtifactDependency and SnapshotDependency

DESCRIPTION:
Add missing Inherited flag for ArtifactDependency and SnapshotDependency

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamCitySharp] - modify last commit string by bool...

DESCRIPTION:

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

Commit:
:art: Change properties name to a good cohesion and initialize the object Credentials in your scope

Commit:
:art: Change properties name to a good cohesion and initialize the object Credentials in your scope

mavezeau committed:
[TeamcitySharp] - Add pin and unpin method by a build number

DESCRIPTION:
Add pin and unpin method by a build number
Reorder method in the Ibuild interface

TEST:
Manual test

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

chaami committed:
Quick and mostly naive fixes of the broken tests.

davidwhitney committed:
NET Standard Port - Sadly requires additional dependencies, probably needs the removal of JsonFx as it's relying on an unofficial port.

Commit:
Fix AttachVcsRoot which throws exception from EasyHttp

An unexpected error has occurred: The encoding requested does not have a corresponding decoder

Stacktrace:    at EasyHttp.Codecs.DefaultDecoder.ObtainDeserializer(String contentType) in C:\BuildAgent\work\f8da234c6955617\src\EasyHttp\Codecs\DefaultDecoder.cs:line 44
   at EasyHttp.Codecs.DefaultDecoder.DecodeToStatic[T](String input, String contentType) in C:\BuildAgent\work\f8da234c6955617\src\EasyHttp\Codecs\DefaultDecoder.cs:line 20
   at EasyHttp.Http.HttpResponse.StaticBody[T](String overrideContentType) in C:\BuildAgent\work\f8da234c6955617\src\EasyHttp\Http\HttpResponse.cs:line 115
   at TeamCitySharp.Connection.TeamCityCaller.Post[T](Object data, String contenttype, String urlPart, String accept)
   at TeamCitySharp.Connection.TeamCityCaller.PostFormat[T](Object data, String contenttype, String accept, String urlPart, Object[] parts)
   at TeamCitySharp.ActionTypes.VcsRoots.AttachVcsRoot(BuildTypeLocator locator, VcsRoot vcsRoot)

e-i-n-s committed:
Fix URL for TeamCity REST-API Request

mavezeau committed:
Merge pull request #5 from yurisouza/feature/good-habits-clean-code

Feature/good habits clean code

mavezeau committed:
Merge pull request #16 from lKoTl/fix/VcsRootsAttach

Fix AttachVcsRoot which throws exception from EasyHttp

mavezeau committed:
Merge branch 'master' into master

mavezeau committed:
Merge pull request #17 from e-i-n-s/master

Fix '+' sign in URL

mavezeau committed:
[TeamCityCaller] -  Fix issue after merge

DESCRIPTION:

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamCitySharp] - Publish new version and cleanup code

DESCRIPTION:

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
Merge pull request #11 from chaami/fix_tests

Quick and mostly naive fixes of the broken tests.

mavezeau committed:
Merge branch 'pr/14' into DotNetCore

mavezeau committed:
[DotNetCore] - Fix test and cleanup

DESCRIPTION:

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[DotNetCore] - Remove JSonFx and use only the newton.json

DESCRIPTION:
Remove JSonFx and use only the newton.json
Code cleanup

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[DotNetCore] - Fix tests and post and put command

DESCRIPTION:

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
Update README.md

owena1337 committed:
Added VcsBranchName to get the branch name.

mavezeau committed:
[TeamcitySharp] - Bug fix with post command

DESCRIPTION:
Bug fix with post command

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamcitySharp] - Fix bug #18

DESCRIPTION:
Add missing section when teamcty return only a href for a section
https://github.com/mavezeau/TeamCitySharp/issues/18

TEST:
Manual and integration added

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamcitySharp] - Add UseVersion to specify the rest api required

DESCRIPTION:
Add UseVersion to specify the rest api required

example:
var client = new TeamCityClient("localhost:81");
    client.Connect("admin", "qwerty");
    client.UseVersion("7.0"); // 6.0, 7.0, 8.1, 9.0, 9.1, 10.0, 2017.1, 2017.2, latest

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamCitySharp] - simple fix for big extract

DESCRIPTION:

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamcitySharp] - Import new DomainEntities and new fields

DESCRIPTION:
Import new DomainEntities and new fields

TEST:
Manual Tests

New DomainEntities:
 - BuildChange(s)
 - Datas
 - Entries
 - Entry
 - Link(s)
 - MetaData
 - Mute(s)
 - Problem(s)
 - ProblemOccurrence(s)
 - ProblemScope
 - ProblemTarget
 - TestOccurrence(s)

Modified DomainEntities

 - Agents
 - AgentWrapper
 - Build
 - CompatibleAgents
 - Test
 - Triggered

New Fields linked to DomainEntities

Added new feature to rename a fieldId for a group

REVIEWER:
Carlo

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamCitySharp] -  Fix README.md and add missing GetFields for BuildInvestigation and Users

DESCRIPTION:
Fix README.md and add missing GetFields for BuildInvestigation and Users

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamCitySharp] - Build fix ... missing to save a file after review...

DESCRIPTION:

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

Commit:
Renamed CopyProject parameters to be more explicity about source and destination

Commit:
Id is a valid value for setting VscRootValue

tusmanakhter committed:
[TeamcitySharp] - Added support for new templates api endpoints

DESCRIPTION:
Support for templates api endpoint (Version >= TeamCity 2017.2)
Old endpoint ended in /template
New endpoint ends in /templates

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
Merge pull request #24 from tusmanakhter/master

[TeamcitySharp] - Added support for new templates api endpoints

mavezeau committed:
Merge pull request #23 from SonnyCampbell/feature/add-id-to-vcsrootvalue

Id is a valid value for setting VscRootValue

mavezeau committed:
Merge pull request #21 from SonnyCampbell/feature/renaming-project-copy-params

Clearer CopyProject parameters

mavezeau committed:
Merge pull request #20 from owena1337/patch-1

Added VcsBranchName to get the branch name.

mavezeau committed:
[TeamCitySharp] - Bump revision before create a new version in nuget version 2.0.5

DESCRIPTION:

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamCitySharp] - Fix statistics data, now it`s possible to fetch stats directly from the build object (with field)

DESCRIPTION:
Fix statistics data, now it`s possible to fetch stats directly from the build object (with field)

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamCitySharp] -  Create version 2.0.6

DESCRIPTION:

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

tidharmor committed:
Check if body is null

using client.ServerInformation.TriggerServerInstanceBackup calls Post with null body which causes an exception.

tusmanakhter committed:
[TeamcitySharp] - Fixed issue preventing artifact downloads

DESCRIPTION:
Artifact download capability restored
Added a test for artifact download

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
Merge pull request #27 from tusmanakhter/master

[TeamcitySharp] - Fixed issue preventing artifact downloads

tusmanakhter committed:
[TeamcitySharp] - Add method to create a build configuration with other properties

DESCRIPTION:
Build configuration can now be created with all properties instead of adding them after creation.

mavezeau committed:
Merge pull request #28 from tusmanakhter/master

[TeamcitySharp] - Add method to create a build configuration with other properties

mavezeau committed:
[TeamcitySharp] - Fix download artifacts with a git branch

DESCRIPTION:
Fix download artifacts with a git branch

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[TeamCitySharp] - New version 2.0.8

DESCRIPTION:
Added support to specify branch when you need to download an artifact

TEST:
Manual

REVIEWER:
Tusman

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

oubenal committed:
[TeamCitySharp] -  Support test occurrences

DESCRIPTION: Add new feature to get test occurrences
By build/project/test locator

oubenal committed:
[TeamCitySharp] - Fix typo

DESCRIPTION:
Misspelling of investigation

mavezeau committed:
Merge pull request #30 from oubenal/fix/typo

[TeamCitySharp] - Fix typo

mavezeau committed:
Merge pull request #29 from oubenal/feature/testoccurences

[TeamCitySharp] -  Add support for Test Occurences

mavezeau committed:
Merge pull request #26 from tidharmor/master

Check if body is null

mavezeau committed:
[TeamcitySharp] - Add support to fetch branch information by project or build config

DESCRIPTION:
Add support to fetch branch information by project or build config
Add missing field for branches

new Method
 + BuildConfigs
    GetBranchesByBuildConfigurationId(string buildTypeId, BranchLocator locator = null)

 + Project
    GetBranchesByBuildProjectId(string projectId, BranchLocator locator = null)
TEST:

REVIEWER:
Tusman

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
[Teamcity] - Bump version to 2.0.9

DESCRIPTION:

TEST:

REVIEWER:

REFERENCES:
-TP Task: None
-Defect: None
-Wiki: None

mavezeau committed:
Convert project to 2019

Added new connection feature with access token (since Teamcity 2019.1)

reviewer
Tusman

mavezeau committed:
Create version 2.0.10

mavezeau committed:
New Version 2.0.11

Fix previous release 2.0.10

Rename CheckForUserNameAndPassword to CheckForAuthRequest to support token connection

mavezeau committed:
Add new feature for issue #31

Add new method
 # void PutRawBuildStep(BuildTypeLocator locator, string rawXml);
 # BuildSteps GetRawBuildStep(BuildTypeLocator locator);

https://github.com/mavezeau/TeamCitySharp/issues/31

mavezeau committed:
Fix instability with download artifact.

mavezeau committed:
New version 2.0.12

mavezeau committed:
Fix issue with big artifact download

Added support  of .netcore 3.1

mavezeau committed:
create a new version 2.0.14

mavezeau committed:
reverted modification on de 2.0.14

mavezeau committed:
Create new version 2.0.15

mavezeau committed:
Fix issue with download a big file

mavezeau committed:
New version 2.0.16

droyad committed:
Added DefaultFilter parameter

droyad committed:
Added more filters

mavezeau committed:
Merge pull request #35 from droyad/defaultFilter

Added some more filters

mavezeau committed:
Fix project compilation since last contribution

mavezeau committed:
Create build 2.0.17
